src/agent/communicator/src/communicator.cpp

```diff
@@ -38,11 +39,21 @@ namespace communicator
 
             m_retryIntervalSecs = std::stol(getStringConfigValue("agent", "retry_interval_secs"));
         }
+
+        if (getHeaderInfo != nullptr)
+        {
+            m_getHeaderInfo = std::move(getHeaderInfo);
+        }
+        else
+        {
+            LogWarn("Get header info not set.");
```

Comment:
"Perhaps 'Header info not set' is better, otherwise it could be understood as referring to the header of a GET method.